### ISSUE TITLE
TF转化后的样本数据训练时 没有全部用上是什么原因？

### ISSUE THREAD

**Issue by buptzzl** [tobegit3hub/tensorflow_template_application#29]
上千万条转化为TF格式后数据，用sparse_classifier.py 的默认配置训练时不到1分钟就结束了，貌似样本没有读完。这个是什么原因？

**Comment by huyong1109** [comment]
直接上中文？ 作者能看懂吗？ 有报错信息吗？

**Comment by buptzzl** [comment]
谢谢@huyong1109 解答。没有报错信息，我尝试在 sess.run() 里把捕捉的 rs.OutOfRangeError  打印了一下，日志如下：
2017-08-23 20:52:15,268 INFO 399 Step: 100, train loss: 11470849.0, mae: 11470849.0, batch_labels: [[ 0.07468913]], tpred: [[ 11470849.]]
2017-08-23 20:52:15,331 INFO 417 Step: 100 catch OutOfRangeErr: RandomShuffleQueue '_3_shuffle_batch_1/random_shuffle_queue' is closed and has insufficient elements (requested 340, current size 100)
         [[Node: shuffle_batch_1 = QueueDequeueManyV2[component_types=[DT_STRING], timeout_ms=-1, _device="/job:localhost/replica:0/task:0/cpu:0"](shuffle_batch_1/random_shuffle_queue, shuffle_batch_1/n)]]

Caused by op 'shuffle_batch_1', defined at:
  File "sparse_regression.py", line 664, in <module>
    main()
  File "sparse_regression.py", line 154, in main
    min_after_dequeue=MIN_AFTER_DEQUEUE)
  File "/Users/zhouliaoming/anaconda/lib/python3.6/site-packages/tensorflow/python/training/input.py", line 1217, in shuffle_batch
    name=name)
  File "/Users/zhouliaoming/anaconda/lib/python3.6/site-packages/tensorflow/python/training/input.py", line 788, in _shuffle_batch
    dequeued = queue.dequeue_many(batch_size, name=name)
  File "/Users/zhouliaoming/anaconda/lib/python3.6/site-packages/tensorflow/python/ops/data_flow_ops.py", line 457, in dequeue_many
    self._queue_ref, n=n, component_types=self._dtypes, name=name)
  File "/Users/zhouliaoming/anaconda/lib/python3.6/site-packages/tensorflow/python/ops/gen_data_flow_ops.py", line 946, in _queue_dequeue_many_v2
    timeout_ms=timeout_ms, name=name)
  File "/Users/zhouliaoming/anaconda/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py", line 767, in apply_op
    op_def=op_def)
  File "/Users/zhouliaoming/anaconda/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 2506, in create_op
    original_op=self._default_original_op, op_def=op_def)
  File "/Users/zhouliaoming/anaconda/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 1269, in __init__
    self._traceback = _extract_stack()

OutOfRangeError (see above for traceback): RandomShuffleQueue '_3_shuffle_batch_1/random_shuffle_queue' is closed and has insufficient elements (requested 340, current size 100)
         [[Node: shuffle_batch_1 = QueueDequeueManyV2[component_types=[DT_STRING], timeout_ms=-1, _device="/job:localhost/replica:0/task:0/cpu:0"](shuffle_batch_1/random_shuffle_queue, shuffle_batch_1/n)]]

八卦一下：关注了作者的微博，都是中国人:)

**Comment by huyong1109** [comment]
给这个rep的作者点个赞！ 没跑过具体代码， 但是从报错内容可以给出我的推断。 首先， 代码中用string_input_producer和shuffle_batch， 会在threads = tf.train.start_queue_runners(coord=coord, sess=sess)后自动开几个thread来读数据， 报错中说队列已经close了， 所以，要么string_input_producer的输入路径不对， thread没有数据可读， shuffle_batch的queue 没有数据。 还有可能就是你的版本跟作者的版本不同， start_queue_runners的用法改变了， thread都没起起来。

**Comment by huyong1109** [comment]
不好意思，  (requested 340, current size 100)说明，应该数据生成有问题， 或者路径有问题。 queue_runner已经正常跑了。

**Comment by buptzzl** [comment]
恩。输入只有一个大文件，读了一部分数据后没有就终止了

**Comment by buptzzl** [comment]
在run() 中添加  batch_labels, logits, cross_entropy  后解决。原因还不大理解？
